exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -185,7 +185,12 @@ def __init__(
 
         if parsed_url.netloc:
             self._endpoint = parsed_url.netloc
-
+            if parsed_url.path:
+                logger.warning(
+                    f"Endpoint set to {self._endpoint}, which differs from the "
```

Comment:
`self._endpoint` and `endpoint` can have the same value and this warning would be shown, right? These changes need tests to make sure that we are not showing a warning when it is not necessary.